File: frontend/src/types/session.ts
export interface SessionListItem {
  id: string;
  title: string | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface MessageRead {
  id: string;
  role: string;
  content: string;
  message_type: string | null;
  metadata_: Record<string, unknown> | null;
  sequence_num: number;
  created_at: string;
}

export interface SessionRead {
  id: string;
  title: string | null;
  agent_config: Record<string, unknown> | null;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  messages: MessageRead[];
}

export interface ChatRequest {
  message: string;
}

export type SSEEventType = "thinking" | "execute" | "solution" | "error" | "done";

export interface SSEEvent {
  event: SSEEventType;
  data: Record<string, unknown>;
}

export interface ChatMessage {
  role: "user" | "assistant";
  content: string;
  type: SSEEventType | "user";
}

File: backend/app/schemas/session.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class SessionCreate(BaseModel):
    title: str | None = None


class MessageRead(BaseModel):
    id: uuid.UUID
    role: str
    content: str
    message_type: str | None = None
    metadata_: dict | None = None
    sequence_num: int
    created_at: datetime

    model_config = {"from_attributes": True}


class SessionRead(BaseModel):
    id: uuid.UUID
    title: str | None
    agent_config: dict | None
    is_active: bool
    created_at: datetime
    updated_at: datetime
    messages: list[MessageRead] = []

    model_config = {"from_attributes": True}


class SessionListItem(BaseModel):
    id: uuid.UUID
    title: str | None
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ChatRequest(BaseModel):
    message: str = Field(..., min_length=1, max_length=32768)
